### ISSUE TITLE
auth:export command returns duplicate results

### ISSUE THREAD

**Issue by jpreynat** [firebase/firebase-tools#1635]
### [REQUIRED] Environment info

**firebase-tools:** 7.3.0 and earlier

**Platform:** macOS

### [REQUIRED] Test case

```bash
$ firebase-tools auth:export auth.json

# Log number of accounts in the file (found by their `localId` property)
$ cat auth.json | grep localId | wc -l
3949724

# Log actual number of de-duplicated accounts in the file
$ cat auth.json | grep localId | sort | uniq | wc -l
367793
```

Our Firebase Auth counts more than 300000 accounts. We export them using the `auth:export` command regularly to analyze them.
Until recently, it worked great. But as you can see above, today the export file includes 10x more accounts than their actual number. The number of de-duplicated accounts matches the number of accounts displayed in our Firebase Auth dashboard, so we are sure that this is the correct one.

After a short investigation, it seems that the Google API that `firebase-tools` relies on to list the accounts `/identitytoolkit/v3/relyingparty/downloadAccount` has changed its behavior and returns a [`nextPageToken`](https://github.com/firebase/firebase-tools/blob/master/src/accountExporter.js#L161) randomly now when the end of the users list is reached.

Here are some logs of an homemade module that uses this API and logs the `nextPageToken` value after each completion call:
```
nextPageToken undefined
Exported 773 users, total: 0
nextPageToken undefined
Exported 773 users, total: 773
nextPageToken undefined
Exported 773 users, total: 1546
nextPageToken undefined
Exported 773 users, total: 2319
nextPageToken undefined
Exported 773 users, total: 3092
nextPageToken undefined
Exported 773 users, total: 3865
nextPageToken undefined
Exported 773 users, total: 4638
nextPageToken undefined
Exported 773 users, total: 5411
nextPageToken visitor_uid
Exported 773 users, total: 6184
nextPageToken undefined
Finished with 6957
```

You can see that at some point, `nextPageToken` is `visitor_uid`, which ends the loop, while it returns `undefined` most of the time.

Given the code of the `auth:export` function, it's logic that the command currently loops, potentially endlessly.
We should add a check [here](https://github.com/firebase/firebase-tools/blob/master/src/accountExporter.js#L161-L162) to ensure that `nextPageToken` has actually been returned with a value before iterating again.

@mbleigh Could you also check the reason of this update on the Google API? Maybe it's just a temporary bug, but since the API is deprecated, it's not possible to check for its current status.

### [REQUIRED] Steps to reproduce

See above.

### [REQUIRED] Expected behavior

The file should include only the correct number of accounts from Firebase Auth.

### [REQUIRED] Actual behavior

The file includes duplicates of the accounts, and `firebase-tools` can potentially iterate endlessly on calling the `identitytoolkit` endpoint.

**Comment by jpreynat** [comment]
PR to fix the issue is here #1636

**Comment by PeteX** [comment]
Thanks for working to address this issue.  Perhaps I'm doing something wrong, but here it seems to have made the problem worse.  I'm running this command:

`firebase auth:export --project [removed] firebase-auth-export.json --format=json`

Instead of exporting accounts multiple times, the export file now only contains a single account!  It seems to be the account that is first when sorted by uid.

**Comment by jpreynat** [comment]
Indeed, it seems that there is an issue with how the accounts are written to the file.
I experience the same issue, even though after logging a few infos, the Google API is returning the correct list of users and the `_writeUsersToFile` method is correctly iterating over them.
It looks like there's an issue with the end of the writing process.

**Comment by jpreynat** [comment]
After further investigation, this is happening because the data written to the stream for the output file doesn't get to be flushed before the process ends.
It should have worked before because the `serialExportUsers` made another request to the API with empty results, and this gave the process enough time to write the buffered data to the file.
Since the method now returns early, the buffered data doesn't have enough time to be flushed to the file.
I'm going to make another PR to fix this.

**Comment by jpreynat** [comment]
Will be fixed by #1642

**Comment by PeteX** [comment]
Thank you @jpreynat—especially because reading the thread above it doesn't sound as though you are a Googler, so it's not really your responsibility!
